server/src/main/java/io/spine/server/event/EventStore.java

```diff
@@ -99,13 +91,48 @@ public void run() {
     }
 
     /**
+     * Appends the passed events to the history of events.
+     *
+     * <p>As storing events is {@linkplain TenantAwareOperation tenant-dependant}, the first Event
```

Comment:
That's not right :)

Instead we need to group the events by `TenantId` and store those separately.

Another option is to validate that they all have the same `TenantId`.